openedx/adg/lms/applications/serializers.py

```diff
@@ -0,0 +1,105 @@
+"""
+Serializers for applications app
+"""
+from rest_framework import serializers
+
+from .helpers import check_validations_for_current_record, check_validations_for_past_record
+from .models import Education, WorkExperience
+
+
+class EducationSerializer(serializers.ModelSerializer):
+    """
+    Serializer for the `Education` model
+    """
+
+    def validate(self, attrs):
+        """
+        Custom validations on education attrs
+
+        Arguments:
+            attrs (dict): Dictionary containing education attributes
+
+        Returns:
+            dict: Returns updated education attributes after validation or raises validation exceptions
+        """
+        attrs, errors = self._check_area_of_study_validations(attrs)
+
+        if attrs.get('is_in_progress'):
+            errors.update(
+                check_validations_for_current_record(
+                    attrs,
+                    '{key} isn\'t applicable for degree in progress', 'date_completed_month', 'date_completed_year'
+                )
+            )
+        else:
+            errors.update(
+                check_validations_for_past_record(
+                    attrs, '{key} is required for past degree', 'date_completed_month', 'date_completed_year'
+                )
+            )
+        if errors:
+            raise serializers.ValidationError(errors)
+
+        return attrs
+
+    def _check_area_of_study_validations(self, attrs):
+        """
+        Area of study validations, check if `area_of_study` is provided for all degrees except for `HD` if it isn't
+        raises exception. For `HD` if value of `area_of_study` is provided replace it with empty string`''`.
+
+        Arguments:
+            attrs (dict): Dictionary containing education attributes
+
+        Returns:
+            dicts: Returns updated education attributes and errors if any in dictionary
+        """
+        errors = {}
+        if attrs.get('degree') == 'HD' and attrs.get('area_of_study'):
+            attrs['area_of_study'] = ''
+        elif attrs.get('degree') != 'HD' and not attrs.get('area_of_study'):
+            errors['area_of_study'] = 'area_of_study is required for all degrees above High School Diploma'
```

Comment:
If this error message is visible to the user, can we make `area_of_study ` human-readable